src/components/SpectralProfiler/ProfileFittingComponent/ProfileFittingComponent.tsx

```diff
@@ -0,0 +1,443 @@
+import {SafeNumericInput} from "components/Shared";
+import {ProfileFittingStore} from "stores/ProfileFittingStore"
+import {SpectralProfileWidgetStore} from "stores/widgets";
+import {AppStore} from "stores";
+import {getTimestamp} from "utilities";
+import "./ProfileFittingComponent.scss";
+
+export enum FittingFunction {
+    GAUSSIAN = 0,
+    LORENTZIAN = 1
+}
+
+export enum FittingContinuum {
+    NONE = -1,
+    ZEROTH_ORDER = 0,
+    FIRST_ORDER = 1
+}
+
+export interface ProfileFittingComponentProps {
+    fittingStore: ProfileFittingStore,
+    widgetStore: SpectralProfileWidgetStore
+}
+
+@observer
+export class ProfileFittingComponent extends React.Component<ProfileFittingComponentProps> {
+    @observable isShowingLog: boolean;
+    @observable isShowingResultButton: boolean;
+
+    private onFunctionChanged = (ev) => {
+        this.reset();
+        this.props.fittingStore.setFunction(parseInt(ev.target.value));
+    }
+
+    private onContinuumValueChanged = (ev) => {
+        this.props.fittingStore.setYIntercept(0);
+        this.props.fittingStore.setSlope(0);
+        this.props.fittingStore.setContinuum(parseInt(ev.target.value));
+    }
+
+    private onYInterceptValueChanged = (val: number) => {
+        this.props.fittingStore.setYIntercept(val);
+    }
+
+    private onSlopeValueChanged = (val: number) => {
+        this.props.fittingStore.setSlope(val);
+    }
+
+    private onYInterceptValueLocked = () => {
+        this.props.fittingStore.setLockedYIntercept(!this.props.fittingStore.lockedYIntercept);
+    }
+
+    private onSlopeValueLocked = () => {
+        this.props.fittingStore.setLockedSlope(!this.props.fittingStore.lockedSlope);
+    }
+
+    private cursorSelectingYIntercept = () => {
+        this.props.fittingStore.setIsCursorSelectingYIntercept(!this.props.fittingStore.isCursorSelectingYIntercept);
+    }
+
+    private cursorSelectingSlope = () => {
+        this.props.fittingStore.setIsCursorSelectingSlope(!this.props.fittingStore.isCursorSelectingSlope);
+    }
+
+    private onCenterValueChanged = (val: number) => {
+        this.props.fittingStore.selectedComponent.setCenter(val);
+    }
+
+    private onAmpValueChanged = (val: number) => {
+        this.props.fittingStore.selectedComponent.setAmp(val);
+    }
+
+    private onFwhmValueChanged = (val: number) => {
+        this.props.fittingStore.selectedComponent.setFwhm(val);
+    }
+
+    private onMouseOverResult = () => {
+        this.setIsShowingResultButton(true);
+    }
+
+    private onMouseLeaveResult = () => {
+        this.setIsShowingResultButton(false);
+    }
+
+    private autoDetect = () => {
+        this.props.fittingStore.setHasResult(false);
+        this.props.fittingStore.setComponents(1, true);
+        if (this.props.widgetStore?.plotData?.fittingData) {
+            // const fittingData = this.props.widgetStore.plotData.fittingData;
```

Comment:
@TienHao can we remove these comments? Or if they are to be used later, put a `// TODO` in explaining what they will be used for any why they are currently commented out

Reply:
yes, I forgot to remove them after I finished the feature.